release-notes-1.2.0.md

```diff
@@ -0,0 +1,763 @@
+
+```ballerina
+error? result = from var student in studentList
+ 	where student.score > 1.0
+do {
+        FullName fullName = {firstName: student.firstName, lastName: student.lastName};
+        nameList[nameList.length()] = fullName;
+}
+```
+
+### New streams design
+
+A stream is an object-like value that can generate a sequence of values. There is also a value associated with the completion of the generation of the sequence, which is either nil, indication of the generation of the sequence being completed successfully, or an error.
+
+A stream supports two primitive operations: a next operation and a close operation. The next operation has the same semantics as the next method on the `Iterator` abstract object type. The close operation informs the stream that there will be no more next operations and thus allows the stream to release resources used by the stream.
+
+A stream is iterable. A stream of type `stream<T, C>` has value type `T` and completion type `C`. Calling the next method on the iterator created for an iteration has the same effect as performing the next operation on the stream. The stream does not keep a copy of the sequence of values returned by the next operation. Any subsequent iteration operation on the same stream will not generate further values so the iteration sequence for iterations other than the first will be the empty sequence. 
+
+The stream type provides methods (more precisely functions in the `lang.stream` library) similar to list:
+- map
+- foreach
+- filter
+- reduce
+- iterator
+
+The `map` and `filter` methods return streams and work lazily. Iterable basic types would have a `toStream()` method to convert to a stream; these should handle mutation similarly to iterators; as of now, it supports only for arrays.
+
+```ballerina
+Person[] personList = getPersonList();
+
+stream<Person> personStream = personList.toStream();
+stream<Person> filteredPersonStream = personStream.filter(function (Person person) returns boolean {
+   return person.age > 100 && person.name != "James";
+});
+
+```
+
+### Expression-bodied functions
+
+This release introduces expression-bodied functions whose body is a single expression. The expression function body takes the form `=> E;`, where `E` is any expression. It is equivalent to the block function body, `{ return E; }`. The following is an example where a `Person` record is mapped to an `Employee` record using an expression-bodied function.
+
+```ballerina
+function toEmployee(Person p, string pos) returns Employee => {
+    name: p.fname + " " + p.lname,
+    designation: pos
+};
+```
+
+### Let expressions
+
+A `let` expression allows you to bind variables and evaluate an expression with those variables in the scope.
```

Comment:
```suggestion
This release introduces let expression. It takes the form `T B = E1 in E2`, where E1 is evaluated resulting in a value v. The typed binding pattern T B is matched to v, causing assignments to the variables occurring in B. Then E2 is evaluated with those variables in scope; the resulting value is the result of the let expression.
```

Reply:
@KavinduZoysa @pubudu91 anything else we should add?

Reply:
No this is good. Can you format the code-like parts? e.g., `T B`

Reply:
Edited to format code like parts.